### ISSUE TITLE
contitional render inser node in wrong place

### ISSUE THREAD

**Issue by chasel34** [RobPruzan/react-scratch#2]
When the conditionally rendered node is the first child of the parent node, as shown in the code below, the position of this node is not inserted as expected as the first child of the parent when it changes from an empty node to a real node.

![image](https://github.com/user-attachments/assets/b7d61188-6a50-4f8c-9be9-3d55839f8967)
![image](https://github.com/user-attachments/assets/659a4cdd-ee76-4a45-b525-5a92b1ef84ec)

This seems to be because the 'updateDom' method only concerns itself with the 'lastUpdatedSibling', and the first child node does not have a 'lastUpdatedSibling'.![image](https://github.com/user-attachments/assets/6ad8de1b-94ad-4eca-a96b-74fc401de65f)
![image](https://github.com/user-attachments/assets/21ab13ee-fdea-4bee-bbae-7135963cd293)

**Comment by RobPruzan** [comment]
Yeah, I think you're right. Will try to send up a patch if I get the time
